### PR TITLE
Ci

### PR DESCRIPTION
**Pull request by gizero**
This PR adds some automation to support building on pushes and releasing firmware builds on repository tagging. Still in draft since it lacks a few lines to document what variables need to be set-up for the workflows to properly run. Feel free to review and comment meanwhile.

### CODE REVIEW

**Review by nor2101 — COMMENTED** [review]
Seems fine, we only need to address the BSEC workaround issue

#9

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -22,6 +24,14 @@ jobs:
...
       - name: Build the sketch
         run: make API_SECRET_SALT="${{ secrets.API_SECRET_SALT }}"
+      - name: Zip artifacts
+        run: >
+          zip --junk-paths msp-firmware-${{ env.RELEASE_VERSION }}.zip \
+            README.md \
+            var/build/msp-firmware.ino.bin \
+            var/build/msp-firmware.ino.partitions.bin \
+            var/packages/esp32/hardware/esp32/1.0.4/tools/partitions/boot_app0.bin \
+            var/packages/esp32/hardware/esp32/1.0.4/tools/sdk/bin/bootloader_qio_80m.bin
```
Comment (nor2101) [inline .github/workflows/release.yml]:
```suggestion
            var/packages/esp32/hardware/esp32/1.0.5/tools/partitions/boot_app0.bin \
            var/packages/esp32/hardware/esp32/1.0.5/tools/sdk/bin/bootloader_qio_80m.bin
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        run: curl -o esptool-2.6.1-windows.zip https://dl.espressif.com/dl/esptool-2.6.1-windows.zip && unzip esptool-2.6.1-windows.zip
+      - name: Zip artifacts
+        run: >
+          zip --junk-paths msp-firmware-${{ env.RELEASE_VERSION }}.zip \
+            README.md \
+            scripts/RUNME.BAT \
+            var/build/msp-firmware.ino.bin \
+            var/build/msp-firmware.ino.partitions.bin \
+            var/packages/esp32/hardware/esp32/1.0.4/tools/partitions/boot_app0.bin \
+            var/packages/esp32/hardware/esp32/1.0.4/tools/sdk/bin/bootloader_qio_80m.bin \
```
Comment (nor2101) [inline .github/workflows/release.yml:37]:
```suggestion
            var/packages/esp32/hardware/esp32/1.0.5/tools/partitions/boot_app0.bin \
            var/packages/esp32/hardware/esp32/1.0.5/tools/sdk/bin/bootloader_qio_80m.bin \
```

Comment (gizero) [reply .github/workflows/release.yml]: b80fc1f should have resolved this issue

**Comment by gizero** [comment]
I pushed two more commits to support using `API_SERVER` secret when releasing

**Pull request merged**